protocols/gossipsub/src/behaviour.rs

```diff
@@ -2684,49 +2823,10 @@ where
         }
     }
 
-    // adds a control action to control_pool
-    fn control_pool_add(
-        control_pool: &mut HashMap<PeerId, Vec<ControlAction>>,
-        peer: PeerId,
-        control: ControlAction,
-    ) {
-        control_pool.entry(peer).or_default().push(control);
-    }
-
-    /// Takes each control action mapping and turns it into a message
-    fn flush_control_pool(&mut self) {
-        for (peer, controls) in self.control_pool.drain().collect::<Vec<_>>() {
-            for msg in controls {
-                self.send_message(peer, RpcOut::Control(msg));
-            }
-        }
-
-        // This clears all pending IWANT messages
-        self.pending_iwant_msgs.clear();
-    }
-
-    /// Send a [`RpcOut`] message to a peer. This will wrap the message in an arc if it
-    /// is not already an arc.
-    fn send_message(&mut self, peer_id: PeerId, rpc: RpcOut) {
-        if let Some(m) = self.metrics.as_mut() {
-            if let RpcOut::Publish(ref message) | RpcOut::Forward(ref message) = rpc {
-                // register bytes sent on the internal metrics.
-                m.msg_sent(&message.topic, message.raw_protobuf_len());
-            }
-        }
-
-        self.events.push_back(ToSwarm::NotifyHandler {
-            peer_id,
-            event: HandlerIn::Message(rpc),
-            handler: NotifyHandler::Any,
-        });
-    }
```

Comment:
what is the reasoning for removing these functions? keeping functions shorts generally improves readability

Reply:
Sorry for unresolving this but I was just wondering: couldn't we instead here in `send_message` simply call something like `RpcSender::send_message`?
`RpcSender::send_message` could then match on the `RpcOut` enum variant to decide which channel to send the message on.
The advantages would be:
1. smaller diff in the behavior code
2. could keep the control pool logic to bundle control messages
3. no separate `RpcSender::{graft(), prune(), publish(), ..}` etc methods needed anymore and thus code deduplication. The internal channels are type `RpcOut` anyway.